tdrs-frontend/src/utils/index.js

```diff
@@ -0,0 +1,15 @@
+import axios from 'axios'
+
+let memo = null
+
+const exports = {
```

Comment:
Mind saying a little more about what this utility class is for and why it's needed? Could be a good thing to code comment.

Reply:
This was part of our solution to the CSRF issue. We were not able to figure out why cookies couldn't be read by the frontend in production, and decided instead to expose the CSRF token via the `auth_check` endpoint.

This may not be the solution in the long run, but without some sort of solution, all post requests were going to be blocked.

I agree that it could do with some refactoring, and some commentary. I'll work on that today, and get back to you soon.

Reply:
Ok, I've added a comment, and refactored the code to be more clear. @alexsoble

Reply:
Thanks! I think the code comment here is very helpful. Two quick questions on these changes:

1. I noticed that the `axios` instance used to be created with `withCredentials: true` but that configuration was removed; is that not necessary now that you refactored? 
2. Is this latest frontend build pushed to the dev site? 

Not something we need to implement in this PR, but while reviewing I came across these notes on axios config, they might be useful in the future: https://flaviocopes.com/axios-credentials/.

Reply:
1. the `withCredentials` isn't necessary with the way we are currently setting the csrf token.
2. It is pushed to the dev site.

Reply:
Gotcha, thanks!